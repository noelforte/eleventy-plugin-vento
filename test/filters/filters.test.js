import { runBuild } from '../get-instance.js';
import { expect, test } from 'vitest';

const results = await runBuild(import.meta.dirname, {
	eleventy(configApi) {
		configApi.addFilter(
			'uppercase',
			/** @param {string} content */
			(content) => content.toUpperCase()
		);

		configApi.addFilter(
			'wrapWith',
			(content, tag) => `<${tag}>${content}</${tag}>`
		);

		configApi.addFilter('pageUrlCompare', function (url) {
			return url === this.page.url;
		});
	},
});

test('transform string with built-in method', () => {
	const page = results.find(({ url }) => url === '/basic/');
	expect(page.content).toMatchSnapshot();
});

test('transform string with passed arguments', () => {
	const page = results.find(({ url }) => url === '/with-arguments/');
	expect(page.content).toMatchSnapshot();
});

test('verify page context in scoped data', () => {
	const page = results.find(({ url }) => url === '/with-this/');
	expect(page.content).toBe('true\n');
});
