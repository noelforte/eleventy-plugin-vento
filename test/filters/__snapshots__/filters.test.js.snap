// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`transform string with built-in method 1`] = `
"
Page title is 'A BASIC PAGE'
"
`;

exports[`transform string with passed arguments 1`] = `
"This is a <strong>really cool</strong> filter.
"
`;
